service/matching/matchingEngine.go

```diff
@@ -294,14 +298,15 @@ pollLoop:
 			})
 			if err != nil {
 				// will notify query client that the query task failed
-				completeType := workflow.QueryTaskCompletedTypeFailed
-				e.RespondQueryTaskCompleted(ctx, &m.RespondQueryTaskCompletedRequest{
-					TaskID: common.StringPtr(tCtx.queryTaskInfo.taskID),
-					CompletedRequest: &workflow.RespondQueryTaskCompletedRequest{
-						CompletedType: &completeType,
-						ErrorMessage:  common.StringPtr("server internal error: failed to get nextID " + err.Error()),
-					},
-				})
+				e.deliverQueryResult(tCtx.queryTaskInfo.taskID,
+					&queryResult{err: err, waitNextEventID: mutableStateResp.GetNextEventId()})
```

Comment:
No need to set waitNextEventID on the result

Reply:
will update